cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/shortestpath/SingleSourceShortestPath.scala

```diff
@@ -0,0 +1,44 @@
+package com.twitter.cassovary.algorithms.shortestpath
+
+import com.twitter.cassovary.graph.{Node, GraphDir, BreadthFirstTraverser, DirectedGraph}
+
+case class SingleSourceShortestPath(graph: DirectedGraph[Node], source: Int) extends ShortestPath {
+
+  val bfs = new BreadthFirstTraverser(graph, GraphDir.OutDir, Seq(source))
+  private val (nodes, depths)  = (bfs.toList, bfs.allDepths)
+  private val previous = nodes.map { n =>
```

Comment:
The intent of this map is to keep track of immediate previous neighbors.  I think I had originally tried to use `bfs.prevNbrCounter`but, in all honesty, I found it a little difficult to get the information out of in a way that made sense.  Maybe you could offer some guidance?